agent/connect/ca/provider_vault.go

```diff
@@ -659,14 +659,28 @@ func (v *VaultProvider) setDefaultIntermediateIssuer(vaultResp *vaultapi.Secret,
 		return fmt.Errorf("could not read from /config/issuers: %w", err)
 	}
 	issuersConf := resp.Data
+	defaultIssuer := issuersConf["default"].(string)
+
+	if defaultIssuer == intermediateId {
+		return nil
+	}
+
 	// Overwrite the default issuer
 	issuersConf["default"] = intermediateId
-
 	_, err = v.writeNamespaced(v.config.IntermediatePKINamespace, v.config.IntermediatePKIPath+"config/issuers", issuersConf)
 	if err != nil {
 		return fmt.Errorf("could not write default issuer to /config/issuers: %w", err)
 	}
 
+	// Delete the previously known default issuer to prevent the number of unused
+	// issuers from increasing too much.
+	_, err = v.deleteNamespaced(v.config.IntermediatePKINamespace, v.config.IntermediatePKIPath+"issuer/"+defaultIssuer)
+	if err != nil {
+		v.logger.Warn("Could not delete previous issuer. Manually delete from Vault to prevent the list of issuers from growing too large.",
+			"prev_issuer_id", defaultIssuer,
+			"error", err)
+	}
```

Comment:
Each issuer has a corresponding key I'd probably suggest removing as well, since Consul is managing that and doing cleanup. Keys just take up storage space, they don't really impact things the way issuers (+ CRLs) do... but since Consul is "owning" this mount, I think it is safer for Consul to do so than Vault.

Needs to be issuer removal -> key removal, as keys aren't removable if the using issuer is still present.

Reply:
What's the relationship between keys to issuers? Can I fetch key_id from an issuer or can keys belong to multiple?